tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -434,20 +437,57 @@ func (iptConfigurator *IptablesConfigurator) run() {
 	iptConfigurator.handleInboundIpv4Rules(ipv4RangesInclude)
 	iptConfigurator.handleInboundIpv6Rules(ipv6RangesExclude, ipv6RangesInclude)
 
-	// Execute iptables commands
-	for _, cmd := range iptConfigurator.iptables.BuildV4() {
-		if len(cmd) > 1 {
-			iptConfigurator.ext.RunOrFail(cmd[0], cmd[1:]...)
-		} else {
-			iptConfigurator.ext.RunOrFail(cmd[0])
-		}
+	iptConfigurator.executeCommands()
+}
+
+func (iptConfigurator *IptablesConfigurator) createRulesFile(filePath string, contents string) {
+	f, err := os.Create(filePath)
+	if err != nil {
+		fmt.Println("Unable to create file for iptables-restore")
+		panic(err)
+	}
+	defer f.Close()
+	writer := bufio.NewWriter(f)
+	_, err = writer.WriteString(contents)
+	if err != nil {
+		fmt.Println("Unable to write file for iptables-restore")
+		panic(err)
 	}
+	writer.Flush()
+}
 
-	for _, cmd := range iptConfigurator.iptables.BuildV6() {
+func (iptConfigurator *IptablesConfigurator) executeIptablesCommands(commands [][]string) {
+	for _, cmd := range commands {
 		if len(cmd) > 1 {
 			iptConfigurator.ext.RunOrFail(cmd[0], cmd[1:]...)
 		} else {
 			iptConfigurator.ext.RunOrFail(cmd[0])
 		}
 	}
 }
+
+func (iptConfigurator *IptablesConfigurator) executeCommands() {
+	if iptConfigurator.cfg.RestoreFormat {
+		// Use iptables-restore
+		data := iptConfigurator.iptables.BuildV4Restore()
+		rulesFilePath := filepath.Join("/tmp", fmt.Sprintf("iptables-rules-%d.txt", time.Now().UnixNano()))
```

Comment:
Use [`ioutil.TempFile`](https://golang.org/pkg/io/ioutil/#TempFile) to make sure to use a unique filename.
Same for IPv6 below.

Reply:
thanks.. I was looking for Tempfile method. didn't check in ioutil..